core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/provision/ClusterCreationActions.java

```diff
@@ -153,11 +154,27 @@ protected Selectable createRequest(StackContext context) {
         };
     }
 
+    @Bean(name = "BOOTSTRAPPING_FREEIPA_ENDPOINT_STATE")
+    public Action<?, ?> bootStrappingPrivateEndpointAction() {
+        return new AbstractStackCreationAction<>(BootstrapPublicEndpointSuccess.class) {
+            @Override
+            protected void doExecute(StackContext context, BootstrapPublicEndpointSuccess payload, Map<Object, Object> variables) {
+                clusterCreationService.bootstrapPrivateEndpoints(context.getStack());
```

Comment:
if an exception is thrown here it is not handled

Reply:
I've added a try/catch to the methods being called in `FreeIPAEndpointManagementService`, since what's being called here is just a series of wrappers that ends up in `FreeIPAEndpointManagementService`.